### PR TITLE
[RD] Changes to the Recurring Donation Future Schedule should not create invalid overlapping schedules

### PR DESCRIPTION
**Pull request by npsp-reedestockton**
When Enhanced Recurring Donations is enabled, ensure that there is a maximum of one active schedule with a start date in the future.
# Critical Changes

# Changes

# Issues Closed

# New Metadata

# Deleted Metadata

### CODE REVIEW

**Comment by npsp-reedestockton** [comment]
@force2b @RadSFDeveloper Since the work for this item is ready for review, I thought it made sense to go ahead and create the PR, even though the work item has not made it through our process yet.  Obviously, there is no urgency to review.

**Comment by npsp-reedestockton** [comment]
**lurch: attach W-037430

**Comment by LurchTheButler** [comment]
Tracking <a href="https://foundation.my.salesforce.com/a2x1E000001HoT4QAK" target="_blank">W-037430</a>

**Comment by npsp-reedestockton** [comment]
@sam-knox Could you please review the description of this pull request?

**Comment by npsp-reedestockton** [comment]
@force2b @RadSFDeveloper The work item for this pull request has now been marked Ready for Review

File: src/classes/RD2_RecurringDonations_TDTM.cls [hunk 1]
```diff
@@ -283,7 +283,7 @@ public class RD2_RecurringDonations_TDTM extends TDTM_Runnable {
 
         Map<Id, List<RecurringDonationSchedule__c>> oldSchedules = new Map<Id, List<RecurringDonationSchedule__c>>();
         for (RecurringDonationSchedule__c oldSchedule : [
-                SELECT RecurringDonation__c, EndDate__c FROM RecurringDonationSchedule__c
+                SELECT RecurringDonation__c, StartDate__c, EndDate__c FROM RecurringDonationSchedule__c
```
Comment (force2b) [inline src/classes/RD2_RecurringDonations_TDTM.cls]: I believe this method moved to the schedule service in #4800. I suspect it's not a problem, other than dealing with the eventual merge conflict in this class.

File: src/objects/RecurringDonationSchedule__c.object [hunk 2]
```diff
@@ -157,7 +157,7 @@
         <fullName>Active__c</fullName>
         <description>Establishes whether the Recurring Donation Schedule is active or inactive. Read only</description>
         <externalId>false</externalId>
-        <formula>OR( ISBLANK( EndDate__c ), EndDate__c >= TODAY() )</formula>
+        <formula>OR( ISBLANK( EndDate__c ), AND( EndDate__c >= TODAY(), NOT( StartDate__c > EndDate__c ) ) )</formula>
```
Comment (force2b) [inline src/objects/RecurringDonationSchedule__c.object]: Nice change!

**Comment by Julian88Tex** [comment]
Testing has passed :-)

**Pull request merged**